shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -490,6 +503,59 @@ - (void)dealloc {
   [super dealloc];
 }
 
+#pragma mark - Status Bar touch event related swizzling
+
+static constexpr CGFloat kStandardStatusBarHeight = 20.0;
+
+static IMP __original_sendevent = nil;
+
+void _flutter_sendevent(id self, SEL _cmd, UIEvent *event) {
+  // Call the original implementation.
+  ((void(*)(id,SEL,UIEvent*))__original_sendevent)(self, _cmd, event);
+
+  // Intercept touchBegan events.
+  UITouch *touch = event.allTouches.anyObject;
```

Comment:
Done. Turns out that the correct behaviour is actually that both tap events and drag events that are entirely constrained to the status bar frame trigger scroll. Limiting to taps covers 99% of the expected behaviour.